i6_models/parts/conformer/convolution.py

```diff
@@ -0,0 +1,59 @@
+from torch import nn
+
+
+class ConformerConvolutionV1(nn.Module):
+    """
+    Conformer convolution module.
+    see also: https://github.com/espnet/espnet/blob/713e784c0815ebba2053131307db5f00af5159ea/espnet/nets/pytorch_backend/conformer/convolution.py#L13
+    """
+
+    def __init__(self, channels, kernel_size, dropout=0.1):
+        """
+        :param int channels: number of channels for conv layers
+        :param int kernel_size: kernel size of conv layers
+        """
+        super().__init__()
+
+        # kernel size has to be odd to get same input length when using odd strides. here we use stride 1.
+        assert (kernel_size - 1) % 2 == 0
+
+        self.pointwise_conv1 = nn.Conv1d(
```

Comment:
Why to use `Conv1d` for this and not `Linear`?

Reply:
This is more efficient as you do not need to transpose again, also it seems that logically people treat this as convolution, even if it is basically a simple linear operation.

Reply:
We are transposing anyway. Only the place of transpose changes.

I would also assume that `Linear` is faster than `Conv1d`.

Reply:
From a quick search, `Linear` seems to be indeed faster than `Conv1d` as Albert said. So i think better to use to Linear then. I saw that most pytorch implementations use Conv1d and that's why i used this. But we can use `Linear` if it is faster.